### ISSUE TITLE
Issue: [BUG] Filtering on field with multiple values does not work as expected

### ISSUE THREAD

**Issue by icytree** [estruyf/vscode-front-matter#895]
**Describe the bug**
When trying to filter files with multiple values for a given field, the filter display the combinaison of all the values of the field. See example bellow (screenshots).

**To Reproduce**

1. Create a field (tried with choice multiple and taxonomy)
2. Set up filter for this field
3. Go to dashboard and try to add a filter on this field.

**Expected behavior**
Having the filter consider each value independently.

**Screenshots**

![image](https://github.com/user-attachments/assets/27dc5022-e2c2-49f8-93a7-a00ef96be81a)

![image](https://github.com/user-attachments/assets/f9f05cc7-9055-4b2a-baa2-283189c39817)

**Device:**

- OS: Win10
- Front Matter CMS Version 10.6.0
- Edge

**Comment by estruyf** [comment]
Thanks @icytree for opening the issue.

**Comment by icytree** [comment]
My pleasure! Very nice tool you build here. As a side note, I did some more testing and the issue does not arise for the tags field.

**Comment by estruyf** [comment]
@icytree, I added a fix and it is available for you to test in the beta extension version. Feel free to test it out and let me know how it works.

**Comment by icytree** [comment]
Yep works great! Thank you very much! (idem for the new custom group by feature)
